### DISCUSSION TITLE
Search in Realtime Database

### DISCUSSION THREAD

**Discussion by kazuser** [SchneiderInfosystems/FB4D#75] [Q&A] [answered]
Hi! Many thanks for FB4D! 👍

Could you please explain how to implement search in Realtime Database with FB4D?

 

Thanks in advance.

**Comment by SchneiderInfosystems** [comment, marked as answer]
As an introduction, I would like to point out that compared to a SQL data bank the RTDB offers almost no search functions, and the newer Firestore only limited possibilities for queries.

In the RTDB you can address each node individually and the structure of your data, therefore, becomes very important so that the frequent queries run optimally and with the smallest possible bandwidth.
Additionally, there is the possibility with "OrderBy" and "Limit to first" a simple logic for the search for a field. Have a close look at the Sample/Intro/DemoFB4D.dpr under RTDB>Get>Options.

If you have more complex structures, I recommend you to switch to Firestore. There you have significantly more possibilities with RunQuery. Since I was able to implement a listener on the Firestore, there is no reason to use the RT-DB for more complex data structures.
